### PR TITLE
10 add development section in readme to provide starting point for developers

### PR DESCRIPTION
**Pull request by marrip**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,66 @@
...
+
+The following packages are recommended to be used for the indicated purposes. We are open to discuss any of the following decisions depending on cost and benefit of the switch. If new purposes arise we will expand the list. Prior to suggesting new packages, please do some research on what people recommend and if the suggested package is actually maintained and fairly adopted by the community.
+
+purpose | package
+---|---
+cli | [typer](https://typer.tiangolo.com/)
+linting | [isort](https://pycqa.github.io/isort/), [ruff](https://docs.astral.sh/ruff/)
+logging | [logging](https://docs.python.org/3/library/logging.html)
+plotting | [plotnine](https://plotnine.org/)
+read csv/tsv tables | [polars](https://docs.pola.rs/)
```
Comment (tinavisnovska) [inline README.md:30]: do you want to use `polars` regardless of how small or big the csv file is or is the `csv` package also ok? If it is ok, then does it make sense to add or you think that is obvious?

Comment (marrip) [reply README.md:30]: I would opt for using polars regardless of the file size. Or do you see any advantage with using the csv package like a functionality that polars does not provide?

Comment (tinavisnovska) [reply README.md:30]: So far, i did not came across anything that cannot be done with polars. 

I think my question comes from the habit of avoiding external packages when there is similar functionality already available. However, given that polars is imported anyway, it is better to keep the code consistent and stick to polars if it can provide the needed functionality so i will adjust :)

Comment (marrip) [reply README.md:30]: I agree, sounds good 🙂

**Review by tinavisnovska — APPROVED** [review]
reads well, covers everything from my perspective

File: README.md [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+1. Functions should be named with snake_case. All input variables and output should be [typed](https://docs.python.org/3/library/typing.html). Include a short description under the function definition line. Also provide comments to describe individual sections of the function and for parts that are generally more complex.
+1. Classes should be named with [PascalCase](https://stringcase.org/cases/pascal/). The class should contain a description and a list over all attributes. It also needs a constructor method `__init__`. The same guidelines listed for functions should be applied to methods.
+1. Import your subpackage to `src/tsoppy/cli.py` like so: `from tsoppy.<subpackage>.<module> import <class or function>`. Subsequently, connect your subpackage to a command like so:
+
+    ```python
+    @app.command()
+    def <subpackage name>():
+    ```
+
+1. Provide unit tests in `tests/test_<subpackag>_<module>.py`. If test data is necessary add it under `tests/test_data/<subpackage>_<module>/<test case name>`. Cover all edge cases as well as use cases from the different nodes.
```
Comment (molversmyr) [inline README.md]: A typo is the only thing I can find, the rest looks good to me😁 

```suggestion
1. Provide unit tests in `tests/test_<subpackage>_<module>.py`. If test data is necessary add it under `tests/test_data/<subpackage>_<module>/<test case name>`. Cover all edge cases as well as use cases from the different nodes.
```

Comment (molversmyr) [reply README.md]: Unless of course, you wish to add an ASCII directory structure diagram 😅 
Something like this:

```
.
├── .devcontainer/
│   ├── .devontainer.json
│   └── Dockerfile
├── .github
├── .gitignore
├── src/
│   └── tsoppy/
│       ├── __init__.py
│       ├── cli.py
│       └── <subpackage>/
│           ├── __init__.py
│           └── main.py
├── tests/
│   ├── test_data/
│   │   └── <subpackage>_module/
│   │       └── <test case name>.py
│   └── test_<subpackage>_<module>.py
├── CODEOWNERS
├── Dockerfile
├── LICENSE
├── pyproject.toml
└── README.md
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,66 @@
...
+- `.github` - github action scripts to perform CI related tasks such as testing, linting and building
+- `src/tsoppy` - main package for tsoppy containing code for cli
+- `src/tsoppy/<subpackage>` - package containing a subfunctionality performing a specific task, typically linked to a subcommand
+- `tests` - unit tests and test data
+- `.gitignore` - list of files to be excluded from git
+- `CODEOWNERS` - specifying who is responsible
+- `Dockerfile` - build recipe for docker
+- `LICENSE` - license that we agreed on in InPreD bioinfo group
+- `pyproject.toml` - project configuration, dependencies etc.
+- `README.md` - documentation
```
Comment (marrip) [inline README.md]:
```suggestion
```txt
.
├── .devcontainer/ # contains instructions to build and start a [development container](https://containers.dev/)
│   ├── .devontainer.json
│   └── Dockerfile
├── .github # github action scripts to perform CI related tasks such as testing, linting and building
├── .gitignore # list of files to be excluded from git
├── src/
│   └── tsoppy/ # main package for tsoppy containing code for cli
│       ├── __init__.py
│       ├── cli.py
│       └── <subpackage>/ # package containing a subfunctionality performing a specific task, typically linked to a subcommand
│           ├── __init__.py
│           └── main.py
├── tests/ # unit tests and test data
│   ├── test_data/
│   │   └── <subpackage>_module/
│   │       └── <test case name>.py
│   └── test_<subpackage>_<module>.py
├── CODEOWNERS # specifying who is responsible
├── Dockerfile # build recipe for docker
├── LICENSE # license that we agreed on in InPreD bioinfo group
├── pyproject.toml # project configuration, dependencies etc.
└── README.md # documentation
```

File: README.md (hunk 2 above)
Comment (marrip) [reply README.md]: I liked the idea so I included it in the repo structure section 👍

**Comment by marrip** [comment]
> reads well, covers everything from my perspective

cheers, Tina! 🙏

File: README.md [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
 
 TSO500 v2 post processing cli
+
+## Development
+
+If you are contributing to the codebase of tsoppy, we would like for you to read this section to familiarize yourself with the structure and guidelines of the project.
+
+### Recommended workflow
+
+1. Create an [issue](https://github.com/InPreD/tsoppy/issues/new) which is assigned to yourself, labeled as `enhancement` and of type `feature`. Include a adequate description and link to any relevant old TSOPPI code or other documents.
```
Comment (danielvo) [inline README.md]: I wonder whether "a adequate description" should be "an adequate description" instead, but I am not a native speaker.. No other comments!

**Review by danielvo — APPROVED** [review]
Very helpful having all of this in one place. Thank you very much, Martin!

File: README.md (hunk 4 above)
Comment (marrip) [reply README.md]: you are 100% correct 😉 
```suggestion
1. Create an [issue](https://github.com/InPreD/tsoppy/issues/new) which is assigned to yourself, labeled as `enhancement` and of type `feature`. Include an adequate description and link to any relevant old TSOPPI code or other documents.
```

**Review by molversmyr — APPROVED** [review]
